yjit/src/codegen.rs

```diff
@@ -2445,37 +2445,82 @@ fn gen_definedivar(
     jit: &mut JITState,
     ctx: &mut Context,
     asm: &mut Assembler,
-    _ocb: &mut OutlinedCb,
+    ocb: &mut OutlinedCb,
 ) -> CodegenStatus {
+    // Defer compilation so we can specialize base on a runtime receiver
+    if !jit.at_current_insn() {
+        defer_compilation(jit, ctx, asm, ocb);
+        return EndBlock;
+    }
+
     let ivar_name = jit.get_arg(0).as_u64();
     let pushval = jit.get_arg(2);
```

Comment:
This is the value that is pushed on the stack if the instance variable is defined, so it _is_ literally just a value whose sole purpose is to be pushed on the stack. In practice it will always be the string `"instance-variable"`.

So I think the variable name is right - or at least I cannot think of a better name right now. But I'll be happy to add a comment.

Reply:
Done. Let me know if you or somebody else thinks of a better name for the variable - I'll be happy to change it! Naming is hard.

Reply:
Could be a separate PR, but why not push just the boolean `true` instead of the string "instance-variable"? Would be more efficient (no need for a 64-bit GC referenced object)

Reply:
I think compile.c will swap the string for `Qtrue` if the value is only used as a boolean.

Reply:
In this case the string is an argument for your instruction though?

Reply:
> why not push just the boolean `true` instead of the string "instance-variable"?

That would imply a breaking change to Ruby.

```
$ ruby -e '@foo = 1; puts defined?(@foo)'
```
will print "instance-variable". I guess almost all use-cases for `defined?` just check "truthiness" of the result, but there's probably also _lots_ of code that looks at the returned string.

> In this case the string is an argument for your instruction though?

Yes, and that makes it much less of an issue. It's a frozen String, and the same String is pushed every time. So it may _technically_ be a GC-referenced object (I have no idea about that), but it's only a single object across the whole program.